keanu-project/src/main/java/io/improbable/keanu/distributions/tensors/continuous/TensorMultivariateGaussian.java

```diff
@@ -0,0 +1,35 @@
+package io.improbable.keanu.distributions.tensors.continuous;
+
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+
+public class TensorMultivariateGaussian {
+
+    private TensorMultivariateGaussian() {
+    }
+
+    public static DoubleTensor sample(DoubleTensor mu, DoubleTensor covariance, GaussianVertex gaussianVariates, KeanuRandom random) {
+        final DoubleTensor choleskyCov = covariance.choleskyDecomposition();
+        final DoubleTensor variateSamples = gaussianVariates.sample(random);
```

Comment:
You could use `random.nextGaussian(mu.getShape())` instead of GaussianVertex gaussianVariates. This removes the need for a GaussianVertex in the MultivariateGaussianVertex too.

Reply:
Ah much nicer, thanks. Changed.